### DISCUSSION TITLE
Spotify iOS SDKで曲を流す方法について

### DISCUSSION THREAD

**Discussion by konnyakuken** [lit-iphone-course/lit-iphone-course#51] [Question-original] [answered]
(1) 実現したいこと
[こちらのサイト](https://note.com/npaka/n/ne4dd92c92e1c)のように一旦ボタンを押すとSpotifyから曲を再生したい
(2) 実現するために自分が試した内容とその結果
上記のサイトを参考にしつつセットアップを進めたところ、`var appRemote: SPTAppRemote!`にて`Cannot find type SPTAppRemote in scope`のスコープ内にそれが無いエラーが出てきます。おそらくセットアップで必要な要素が上手く入ってなくてエラーが出ているのですが、どこが原因かわからない状態です。
(3) 調べた内容や資料
[Spotify公式クイックスタート](https://developer.spotify.com/documentation/ios/quick-start/)や、[Spotify iOS SDKの使い方](https://note.com/npaka/n/ne4dd92c92e1c)を元に設定できていない箇所や命名規則が間違っていないかをチェックしましたが原因がわからず...
(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
多分delegateの設定かSpotifyiOSが上手く設定できていないのかなぁみたいな気がします。
(5) GitHubのリンク
https://github.com/konnyakuken/OriginalMusicApp

**Comment by txxxxc** [comment, marked as answer]
以下の方法試したらコンパイル通ったので共有します！
原因としてはライブラリのインポートがうまくいってなくて、そんなクラス無いって怒られてるっぽいです！

### 1. headerファイルを移動させる
headerファイルを`OriginalMusicApp/originalMusicApp`から`OriginalMusicApp`配下に移動させる(これ自分の環境依存なのかもしれんので違ってそうだったら無視してください)

### 2.plistファイルを弄る
[iOS SDK Quick Start | Spotify for Developers](https://developer.spotify.com/documentation/ios/quick-start/)の`Add Bridging Header`に書いてる以下の手順を行う
>Then you’ll need to set the location of this bridging header by:
In the File Navigator, click on your project.
Click your project under Targets
Go to Build Settings
In the search box, enter Objective-C Bridging Header
Besides Objective-C Bridging Header, double click and enter [YourApp]-Bridging-Header.h

### 3. frameworkからxcframeworkに移行する
frameworkでライブラリを入れるよりも、アーキテクチャの差異を考えずにビルドできるxcframeworkを入れる
sdkのリンク↓
https://github.com/spotify/ios-sdk

参考程度に試してみてくださいー！

**Comment by konnyakuken** [comment]
無事にコンパイル出来ました!!ありがとうございます🙏
